src/ripple/app/ledger/impl/InboundLedgers.cpp

```diff
@@ -292,10 +298,19 @@ class InboundLedgersImp
                 assert (it.second);
                 acquires.push_back (it);
             }
+            for (auto const& it : mRecentFailures)
+            {
+                if (it.second > 1)
+                    ret[beast::lexicalCastThrow <std::string>(
+                        it.second)][jss::failed] = true;
+                else
+                    ret[to_string (it.first)][jss::failed] = true;
+            }
         }
 
         for (auto const& it : acquires)
         {
+            // getJson is expensive, so call without the lock
```

Comment:
This comment isn't really helpful. You open the lock as soon as you've finished the calculations that need it, whether the next operations are expensive or cheap...